activesupport/lib/active_support/inflector/methods.rb

```diff
@@ -69,12 +69,17 @@ def singularize(word, locale = :en)
     def camelize(term, uppercase_first_letter = true)
       string = term.to_s
       if uppercase_first_letter
-        string = string.sub(/^[a-z\d]*/) { |match| inflections.acronyms[match] || match.capitalize }
+        string = string.sub(/^[a-z\d]*/) { |match| inflections.acronyms[match] || match.capitalize! || match }
       else
-        string = string.sub(inflections.acronyms_camelize_regex) { |match| match.downcase }
+        string = string.sub(inflections.acronyms_camelize_regex) { |match| match.downcase! || match }
+      end
+      string.gsub!(/(?:_|(\/))([a-z\d]*)/i) do
+        if $1
+          "::#{inflections.acronyms[$2] || $2.capitalize! || $2}"
+        else
+          inflections.acronyms[$2] || $2.capitalize! || $2
+        end
```

Comment:
```suggestion
        substituted = inflections.acronyms[$2] || $2.capitalize! || $2
        $1 ? "::#{substituted}" : substituted
```

Would it be clearer to highlight the namespacing this way?

Reply:
Probably yes. Let me change that and re-run the benches just in case.

Reply:
Done. Benchmark is the same.